src/mesh/abaqus_io.C

```diff
@@ -852,30 +852,49 @@ void AbaqusIO::read_sideset(std::string sideset_name, sideset_container_t & cont
   dof_id_type elem_id=0;
   unsigned side_id=0;
   char c;
-  std::string dummy;
+  std::string elset_name, dummy;
 
   // Read until the start of another section is detected, or EOF is encountered
   while (_in.peek() != '*' && _in.peek() != EOF)
     {
-      // The strings are of the form: "391, S2"
+      // The strings are either of the form: "391, S2" or "Elset_1, S3"
 
       // Read the element ID and the leading comma
       _in >> elem_id >> c;
+      if (_in)
+      {
+        // For the form of "391, S2"
+        // Read another character (the 'S') and finally the side ID
+        _in >> c >> side_id;
 
-      // Read another character (the 'S') and finally the side ID
-      _in >> c >> side_id;
-
-      // Store this pair of data in the vector
-      id_storage.push_back( std::make_pair(elem_id, side_id) );
+        // Store this pair of data in the vector
+        id_storage.push_back( std::make_pair(elem_id, side_id) );
+      }
+        else
+        {
+            // For the form of "Elset_1, S3"
+            _in.clear();
+            std::getline(_in, elset_name, ',');
+
+            // Read another character (the 'S') and finally the side ID
+            _in >> c >> side_id;
+
+            auto it = _elemset_ids.find(elset_name);
+            if (it == _elemset_ids.end())
+            libmesh_error_msg(
+                "The element set referred to by the surface definition does not exist.");
```

Comment:
If you include "libmesh/utility.h" at the top of the file, you can shorten this bit to:
```
const auto & vec = libmesh_map_find(_elemset_ids, elset_name);
for (const auto & elem_id_in_elset : vec)
  ...
```
Also drop the curly braces for 1-line for loop. I can fix this before merge if you prefer.

Reply:
Thanks for the comment! I'll fix this one.